src/api/Item.js

```diff
@@ -86,20 +86,22 @@ class Item extends Base {
     /**
      * Merges new data with old data and returns new data
      *
-     * @param {String} cacheKey - the cache key of item to merge
-     * @param {String} key - the attribute to merge
-     * @param {Object} value - the value to merge
-     * @return {void}
+     * @param {String} cacheKey - The cache key of item to merge
+     * @param {String} key - The attribute to merge
+     * @param {Object} value - The value to merge
+     * @return {BoxItem} The newly updated object from the cache
      */
-    merge(cacheKey: string, key: string, value: any): void {
+    merge(cacheKey: string, key: string, value: any): BoxItem {
+        const cache: Cache = this.getCache();
+
         if (this.isDestroyed()) {
-            return;
+            return cache.get(cacheKey);
```

Comment:
Not sure if it matters if we return the object here, but I couldn't get flow to play nicely with `return;`

Reply:
You can just remove the destroyed part.
Merge function is synchronous, if the code reaches merge function, then its never destroyed.
Dont think that check will ever be true